src/core/server/http/http_server.ts

```diff
@@ -103,16 +114,23 @@ export class HttpServer {
     const serverOptions = getServerOptions(config);
     this.server = createServer(serverOptions);
 
+    this.setupBasePathRewrite(config);
+
     return {
       options: serverOptions,
       registerRouter: this.registerRouter.bind(this),
-      registerOnRequest: this.registerOnRequest.bind(this),
+      registerOnPreAuth: this.registerOnPreAuth.bind(this),
+      registerOnPostAuth: this.registerOnPostAuth.bind(this),
       registerAuth: <T>(
         fn: AuthenticationHandler<T>,
         cookieOptions: SessionStorageCookieOptions<T>
       ) => this.registerAuth(fn, cookieOptions, config.basePath),
       getBasePathFor: this.getBasePathFor.bind(this, config),
       setBasePathFor: this.setBasePathFor.bind(this),
+      auth: {
```

Comment:
**note**: HttpServer in the current state is mix of infrastructure layer & application layer logic, we can restructure it later, as soon as finish with https://github.com/elastic/kibana/pull/36804 to avoid conflicts